qiskit/synthesis/linear_phase/cnot_phase_synth.py

```diff
@@ -24,21 +24,21 @@
 
 
 def synth_cnot_phase_aam(cnots, angles, section_size=2):
-    """This function is an implementation of the GraySynth algorithm of
+    r"""This function is an implementation of the `GraySynth` algorithm of
     Amy, Azimadeh and Mosca.
 
     GraySynth is a heuristic algorithm from [1] for synthesizing small parity networks.
-    It is inspired by Gray codes. Given a set of binary strings S
-    (called "cnots" bellow), the algorithm synthesizes a parity network for S by
-    repeatedly choosing an index i to expand and then effectively recursing on
-    the co-factors S_0 and S_1, consisting of the strings y in S,
-    with y_i = 0 or 1 respectively. As a subset S is recursively expanded,
-    CNOT gates are applied so that a designated target bit contains the
-    (partial) parity ksi_y(x) where y_i = 1 if and only if y'_i = 1 for all
-    y' in S. If S is a singleton {y'}, then y = y', hence the target bit contains
-    the value ksi_y'(x) as desired.
-
-    Notably, rather than uncomputing this sequence of CNOT gates when a subset S
+    It is inspired by Gray codes. Given a set of binary strings :math:`S`
+    (called ``cnots`` bellow), the algorithm synthesizes a parity network for :math:`S` by
+    repeatedly choosing an index :math:`i` to expand and then effectively recursing on
+    the co-factors :math:`S_0` and :math:`S_1`, consisting of the strings :math:`y \in S`,
+    with :math:`y_i = 0` or 1 respectively. As a subset :math:`S` is recursively expanded,
+    ``cx`` gates are applied so that a designated target bit contains the
+    (partial) parity :math:`\chi_y(x)` where :math:`y_i = 1` if and only if :math:`y'_i = 1` for all
+    :math:`y' \in `S`. If :math:`S` is a singleton :math:`\\{y'\\}`, then :math:`y = y'`, hence the
```

Comment:
Questionable use of "singleton" in a computer-science invaded field if you're asking me 😂 Maybe 
```suggestion
    :math:`y' \in `S`. If :math:`S` contains a single element :math:`\\{y'\\}`, then :math:`y = y'`, hence the
```
(the line width will be off in this suggestion though)